.github/workflows/tests.yml

```diff
@@ -0,0 +1,77 @@
+name: Tests
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  python36:
+    name: Python3.6 tests
+    runs-on: ubuntu-20.04
+
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - postgres: 10
+            python: 3.6
+            toxenv: py36
+            toxdir: psycopg3_c
```

Comment:
I suspect that, after all, it would be better to get rid of tox altogether. Tox has problems with this project because it doesn't have a single setup.py in the root, hence the need of different toxdir. But as far as I see it, toxenv is useless if there is a python version already selected by `python-version`, and toxdir is only needed to build or not build the C extension, because then whether to use the C extension in the test or the pure python is dictated by `PSYCOPG3_IMPL`

So I would simplify the things sort of thus way:

1) have more than one job. Currently you have "python36" as job name: I would have two jobs there such as "tests-python" and "tests-c" (there will be a third, "tests-binary")

I have sketched what I think could work here: https://gist.github.com/dvarrazzo/b60928447011dfde993df37ac46bbc38 I don't know it it can be simplified as there are a lot of repetitions. I haven't fixed how to access the database. I'll add some comments on this file about the changes there...

Reply:
Uhm, if we cannot drop the repetitions it wouldn't even be a silly idea to just have a single job, which would install both Python and C and run e.g.

```
pip install ./psycopg3[tests]
PSYCOPG3_IMPL=python pytest --color yes
pip install ./psycopg3_c
PSYCOPG3_IMPL=c pytest --color yes -m 'not subprocess'
```

with the grid deciding for the Python and Postgres versions. Tests are guaranteed to be independent (order is randomised) so it doesn't matter that the test environment is not isolated.